compiler/rustc_builtin_macros/src/deriving/cmp/partial_eq.rs

```diff
@@ -88,13 +74,9 @@ pub fn expand_deriving_partial_eq(
         push,
     );
 
-    // avoid defining `ne` if we can
-    // c-like enums, enums without any fields and structs without fields
-    // can safely define only `eq`.
-    let mut methods = vec![md!(sym::eq, cs_eq)];
-    if !is_type_without_fields(item) {
-        methods.push(md!(sym::ne, cs_ne));
-    }
+    // No need to generate `ne`, the default suffices, and not generating it is
+    // faster.
```

Comment:
The previous version was implemented in https://github.com/rust-lang/rust/pull/31977 and that PR states:

>Most other cases can not omit ne, because any value field may have a custom
PartialEq implementation.

Did something change since that time? Why is omitting `fn ne` possible now?

Reply:
I'm not 100% certain about it, but I think that comment is just wrong. Some discussion on Zulip [here](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/Shrinking.20derive.20code/near/287011553).

Reply:
Ok, looks like this requirement

>Implementations must ensure that eq and ne are consistent with each other

is pretty old and already existed when https://github.com/rust-lang/rust/pull/31977 was submitted.
I guess `fn ne` can be removed then.